Number_types/include/CGAL/FPU.h

```diff
@@ -487,28 +503,43 @@ inline
 void
 FPU_set_cw (FPU_CW_t cw)
 {
+#ifdef CGAL_ALWAYS_ROUND_TO_NEAREST
+  CGAL_assertion(cw == CGAL_FE_TONEAREST);
```

Comment:
Should we make this a CGAL_expensive_assertion or just CGAL_assertion? Is it redundant with Checked_protect_FPU_rounding?

Reply:
If it is redundant, then remove the assertion. Otherwise, that is a regular expression. There is nothing expensive in comparing two words.

Reply:
I should have made this comment on the previous one, which has CGAL_IA_GETFPCW (in Checked_protect_FPU_rounding we use expensive for that).

Reply:
Hmm.

My reasoning was (a) == is cheap, and (b) there are quite a lot of direct calls to FPU_set_cw().

For example, there's a whole bunch of calls in Apollonius_graph_2/compare_quadratic.h

With the assert we can be confident that only code that expects round-to-nearest will compile given CGAL_ALWAYS_ROUND_TO_NEAREST, which means that code that expects a different rounding mode should be easy to identify and correct incrementally.

Reply:
Thanks for the explanation.